src/client/app/containers/LineChartContainer.ts

```diff
@@ -54,73 +78,75 @@ function mapStateToProps(state: State) {
 					throw new Error('Unacceptable condition: readingsData.readings is undefined.');
 				}
 
-				const dataPoints: ChartPoint[] = _.values(readingsData.readings).map(compressedReadingToChartPoint);
+				// Create two arrays for the x and y values. Fill the array with the data from the compressed readings
+				const xData: string[] = [];
+				const yData: number[] = [];
+				const hoverText: string[] = [];
+				_.values(readingsData.readings).forEach(reading => {
```

Comment:
The meter code explicitly creates the array but this does not. This seems cleaner (a little harder to see data on debug). Whatever is chosen, it should probably be the same way. Also, the meters and groups code is very similar and I wonder if it should be refactored out to create a function to do both. This would mean future changes would automatically happen to both.